src/repositories/accountRepository.ts

```diff
@@ -0,0 +1,29 @@
+import { runSelectQuery, runInsertQuery, QueryResult } from "../lib/mysqlDatabase";
+import { MySQLRepository } from "../lib/MySQLRepository";
+import { AccountEntity } from "../types/entities/AccountEntity";
+import { Identity } from "@npmjs_tdsoftware/subidentity";
+
+class AccountRepository extends MySQLRepository<AccountEntity> {
+    get tableName(): string {
+        return "account";
+    }
+
+    async findAllByChainId(chain_id: number): Promise<AccountEntity[]|undefined> {
+        const query = `SELECT *
+                       FROM ${this.tableName} 
+                       WHERE ${this.tableName}.chain_id=${chain_id}`;
+        return (await runSelectQuery<AccountEntity>(query));
+    }
+
+    async insertOrUpdateAccountsOfIdentities(identities: Identity[], chainId: number): Promise<QueryResult> {
+        const data = [identities.map((identity)=> [identity.basicInfo.address, chainId])];
```

Comment:
@a-oertel data is then an array with one item. That one item is an array containing arrays.. Is that correct?

```javascript
const data = [
    [
        ["{address}", "{chainId}"],
        ["{address}", "{chainId}"],
        ["{address}", "{chainId}"],
        ["{address}", "{chainId}"],
        ["{address}", "{chainId}"],
        ["{address}", "{chainId}"],
        ...
    ]
]
```

Reply:
It is working as expected like that